src/vfs2.c

```diff
@@ -1260,7 +1260,8 @@ static void pgno_ht_insert(uint16_t *ht, uint16_t fx, uint32_t pgno)
 	}
 	/* SQLite uses 1-based frame indices in this context, reserving
 	 * 0 for a sentinel value. */
-	ht[hash % SHM_HT_LEN] = fx + 1;
+	fx++;
+	ht[hash % SHM_HT_LEN] = fx;
```

Comment:
See [this CI job](https://github.com/canonical/dqlite/actions/runs/10088688308/job/27894846340#step:4:229)---the version of GCC on Focal flags the original line for truncating from `int` to `short`. (I guess in more recent GCC this diagnostic has gotten smarter for literals.) Writing it this way avoids that diagnostic; it might not be the clearest expression of intent though.

Reply:
Amazing.